DiagnosticsExtension/Models/ConnectionStringValidator/BlobStorageValidator.cs

```diff
@@ -0,0 +1,196 @@
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.DnsLookupFailed;
+                }
+                else if (e is StorageException)
+                {
+                    if (((StorageException)e).RequestInformation.HttpStatusCode == 401)
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.AuthFailure;
+                    }
+                    else if (((StorageException)e).RequestInformation.HttpStatusCode == 403)
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.Forbidden;
+                    }
+                }
+                else
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.UnknownError;
+                }
+                response.Exception = e;
+            }
+
+            return response;
+        }
+
+        public async Task<TestConnectionData> TestConnectionStringViaAppSetting(string appSettingName, string entityName)
+        {
+            var envDict = Environment.GetEnvironmentVariables();
+            string appSettingClientIdValue, appSettingClientCredValue = null;
+            string value = null;
+            BlobServiceClient client = null;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new BlobServiceClient(value);
+                }
+                catch (ArgumentNullException e)
+                {
+                    throw new EmptyConnectionStringException(e.Message, e);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+            }
+            else
+            {
+                try
+                {
+
+
+                    value = Environment.GetEnvironmentVariable(appSettingName + "__blobServiceUri");
+                    if (string.IsNullOrEmpty(value))
+                    {
+                        value = Environment.GetEnvironmentVariable(appSettingName + "__serviceUri");
+                    }
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + "__clientId");
+                        appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + "__credential");
+                        Uri objuri = new Uri(value);
+                        if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                        {
+                            if (appSettingClientCredValue != "managedidentity")
+                            {
+                                throw new ManagedIdentityException("managedidentitymissed");
```

Comment:
Always define a const string at the class level instead of hard coding a string more than once in the code.

Reply:
This should be "managedIdentityCredentialMissing".  "managedidentitymissed" is inaccurate.

Reply:
fixed